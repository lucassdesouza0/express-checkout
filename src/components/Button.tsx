import React from "react";

// import { Container } from './styles';

const Button: React.FC = ({ children }) => {
  return <button>{children}</button>;
};

export default Button;
