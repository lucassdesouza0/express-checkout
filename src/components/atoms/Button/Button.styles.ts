import styled from "@emotion/styled";

export const Button = styled("button")`
  background-color:
    ${(props) =>
    props.type === "button" ? "#eee" : "#5f2eff"};
  color: ${(props) => (props.type === "button" ? "#5f2eff" : "#eee")};

  :hover {
    background-color:
      ${(props) =>
      props.type === "button" ? "#c8c8c8" : "#3500e1"};
  }

  padding: 10px;
  margin-top: 10px;
  border: none;
  border-radius: 5px;
  font-size: 16px;
  font-weight: bold;
  cursor: pointer;
`;

export default Button;
