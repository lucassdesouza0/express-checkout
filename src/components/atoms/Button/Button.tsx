import React, { ReactNode } from "react";

import SButton from "./Button.styles";

interface ButtonProps {
  className?: string;
  type?: "button" | "submit" | "reset";
  children?: ReactNode | string;
  disabled?: boolean;
  onClick?: () => void;
}

const Button = ({
  className,
  type = "button",
  children,
  onClick,
  disabled = false,
  ...props
}: ButtonProps) => {
  return (
    <SButton
      className={className}
      type={type}
      {...props}
      onClick={onClick}
      disabled={disabled}
      name="button"
    >
      {children}
    </SButton>
  );
};

export default Button;
