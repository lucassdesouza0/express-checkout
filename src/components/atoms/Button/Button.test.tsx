import React from "react";
import { screen, render, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import testRender from "../../../tests/render";

import mock from "../../../tests/__mocks__/product";

import Button from "./Button";

describe("Button", () => {
  test("it should mount", () => {
    testRender(<Button></Button>);

    const button = screen.getByRole("button");

    expect(button).toBeInTheDocument();
  });

  test("it should match snapshot", () => {
    const { asFragment } = render(<Button></Button>);
    expect(asFragment()).toMatchSnapshot();
  });
});
