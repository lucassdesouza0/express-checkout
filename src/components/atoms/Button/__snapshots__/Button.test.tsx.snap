// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button it should match snapshot 1`] = `
<DocumentFragment>
  <button
    class="css-hhtkyx-Button enhitpr0"
    name="button"
    type="button"
  />
</DocumentFragment>
`;
